src/server/frontend_wayland/wl_pointer.cpp

```diff
@@ -110,13 +112,20 @@ void mf::WlPointer::handle_event(MirPointerEvent const* event, WlSurface* surfac
             auto point = Point{mir_pointer_event_axis_value(event, mir_pointer_axis_x),
                                         mir_pointer_event_axis_value(event, mir_pointer_axis_y)};
             auto transformed = surface->transform_point(point);
-            handle_enter(transformed.first, transformed.second);
-            handle_frame();
+            if (transformed)
```

Comment:
Just a style nit: I prefer to restrict the scope of everything (in this case transformed). Vis:
 
```
            if (auto const transformed = surface->transform_point(point))
```
I know it makes little difference here (and won't block on it).